### DISCUSSION TITLE
How to stop the effects auto cycling

### DISCUSSION THREAD

**Discussion by knotbeerdan** [PlummersSoftwareLLC/NightDriverStrip#88] [Q&A] [answered]
This is probably something simple I am overlooking, but has anyone been able to disable the auto effect cycling using the demo project?

I have everything set up and working, I also added a few more effects to the demo project, I also have wifi working and am able to change effects using the web page. But I have tried unchecking the boxes on the web page to try to get it to stay on the effect I would like, but it still continues to cycle through the effects I have set up. I have also tried looking through the code and even tried defining and changing the DEFAULT_EFFECT_INTERVAL but I can't seem to figure out how to get it to stop automatically cycling through the different effects. 
![image](https://user-images.githubusercontent.com/13283237/142077665-1221909d-97a5-4d91-b4cc-f0b8feb33c12.png)

**Comment by xcalibur839** [comment, marked as answer]
If you check the HTML and JS on that page, you can see that there is currently no event handler triggered when the checked state of the checkboxes changes. 

I submitted [PR #89](https://github.com/PlummersSoftwareLLC/NightDriverStrip/pull/89) last night to address this, please check it out and report back how it goes! Do note:
- Disabling an effect that is currently running does not immediately stop the effect, it will continue to run until DEFAULT_EFFECT_INTERVAL has elapsed
- If you only have one effect in AllEffects[] and you disable that effect, the LEDs will appear to freeze in whatever their final state was after DEFAULT_EFFECT_INTERVAL has elapsed. This does not happen when there are multiple effects in AllEffects[] and all of the effects are disabled because of the fading between effects that sets all LEDs to Off/Black (i.e. the final state of the LEDs is Off/Black)

Fixing these issues will require additional changes to effectmanager.h (and possibly spiffswebserver.h, elsewhere).

**Reply by knotbeerdan** [reply]
That seems to have done exactly what I was looking for, thank you very much for the quick response and the help!
